### DISCUSSION TITLE
the data retrieved via useFetch() is not reloading after route change

### DISCUSSION THREAD

**Discussion by Alexxosipov** [nuxt/nuxt#17079] [Questions]
Hey guys! I have a simple page `todo-[id].vue`:
```
<script setup>
import Section from '~~/components/UI/containers/section.vue';
import ButtonLink from '~~/components/UI/buttons/button-link.vue';

const route = useRoute()
const todoId = parseInt(route.params.id)
const todo = await useFetch(`https://jsonplaceholder.typicode.com/todos/${todoId}`)
useHead({
  title: `todo #${todoId} | ${todo.data.value.title}`,
  meta: {
    name: 'description',
    content: todo.data.value.title
  }
})
</script>

<template>
  <Section>
    <h1>test page.</h1>
    <div class="mt-4 flex space-x-4">
      <ButtonLink :to="{name: 'todo-id', params: {id: todoId - 1}}" type="secondary" v-show="todoId !== 1">previous</ButtonLink>
      <ButtonLink :to="{name: 'todo-id', params: {id: todoId + 1}}">next</ButtonLink>
    </div>
    <div class="mt-6">
      <p class="text-lg" :class="[todo.data.value.completed ? 'text-green-600' : 'text-red-600']">{{ todo.data.value.title }} (#{{ todo.data.value.id }})</p>
    </div>
  </Section>
</template>
```

Everything is ok on the first hit, but it does not work on route change on the client side. 

Steps to reproduce: 
1. go to /todo-1
2. click on "next" button
3. page url has changed to /todo-2, but the data is still as on the /todo-1 page

What is the trick with useFetch and how to fetch the data properly both on server side and client side? 

P.S. ButtonLink is a simple wrapper for NuxtLink but with styles.

**Comment by Flanker72** [comment]
As written in https://v3.nuxtjs.org/guide/features/data-fetching/#refreshing-data - there is a different approach on watchable query. You need to define `useFetch` URL as function like this:
```ts
const todo = await useFetch(() => `https://jsonplaceholder.typicode.com/todos/${todoId}`)
```
Then your query becomes reactive and it will be updated and refetched after `todoId` changes.
